In [125]:
%run iJO1366_V0.ipynb


[notice] A new release of pip is available: 23.3 -> 23.3.1
[notice] To update, run: pip install --upgrade pip
Note: you may need to restart the kernel to use updated packages.
Maximize
1.0*BIOMASS_Ec_iJO1366_core_53p95M - 1.0*BIOMASS_Ec_iJO1366_core_53p95M_reverse_5c8b1
10fthf_c 10-Formyltetrahydrofolate
12dgr120_c 1,2-Diacyl-sn-glycerol (didodecanoyl, n-C12:0)
12dgr140_c 1,2-Diacyl-sn-glycerol (ditetradecanoyl, n-C14:0)
12dgr141_c 1,2-Diacyl-sn-glycerol (ditetradec-7-enoyl, n-C14:1)
12dgr160_c 1,2-Diacyl-sn-glycerol (dihexadecanoyl, n-C16:0)
12dgr161_c 1,2-Diacyl-sn-glycerol (dihexadec-9-enoyl, n-C16:1)
12dgr180_c 1,2-Diacyl-sn-glycerol (dioctadecanoyl, n-C18:0)
12dgr181_c 1,2-Diacyl-sn-glycerol (dioctadec-11-enoyl, n-C18:1)
12ppd__R_c (R)-Propane-1,2-diol
12ppd__S_c (S)-Propane-1,2-diol
13dpg_c 3-Phospho-D-glyceroyl phosphate
14dhncoa_c 1,4-dihydroxy-2-napthoyl-CoA
14glucan_c 1,4-alpha-D-glucan
15dap_c 1,5-Diaminopentane
1ddecg3p_c 1-dodecanoyl-sn-glycerol 3-phosphate
1hdec9eg3p_c 1

In [126]:
model.medium

{'EX_co2_e': 1000.0,
 'EX_cobalt2_e': 1000.0,
 'EX_glc__D_e': 10.0,
 'EX_h_e': 1000.0,
 'EX_h2o_e': 1000.0,
 'EX_k_e': 1000.0,
 'EX_cu2_e': 1000.0,
 'EX_mg2_e': 1000.0,
 'EX_mn2_e': 1000.0,
 'EX_mobd_e': 1000.0,
 'EX_na1_e': 1000.0,
 'EX_nh4_e': 1000.0,
 'EX_ca2_e': 1000.0,
 'EX_cbl1_e': 0.01,
 'EX_ni2_e': 1000.0,
 'EX_o2_e': 1000.0,
 'EX_cl_e': 1000.0,
 'EX_pi_e': 1000.0,
 'EX_zn2_e': 1000.0,
 'EX_sel_e': 1000.0,
 'EX_slnt_e': 1000.0,
 'EX_so4_e': 1000.0,
 'EX_tungs_e': 1000.0,
 'EX_fe2_e': 1000.0,
 'EX_fe3_e': 1000.0}

In [89]:
model.reactions

[<Reaction EX_cm_e at 0x7efe2e1637c0>,
 <Reaction EX_cmp_e at 0x7efe2e1636d0>,
 <Reaction EX_co2_e at 0x7efe2e163a00>,
 <Reaction EX_cobalt2_e at 0x7efe2e163a90>,
 <Reaction DM_4crsol_c at 0x7efe2e163be0>,
 <Reaction DM_5drib_c at 0x7efe2e163d00>,
 <Reaction DM_aacald_c at 0x7efe2e163e20>,
 <Reaction DM_amob_c at 0x7efe2e163f10>,
 <Reaction DM_mththf_c at 0x7efe2e1637f0>,
 <Reaction EX_colipa_e at 0x7efe2e163610>,
 <Reaction DM_oxam_c at 0x7efe2e1639a0>,
 <Reaction EX_glc__D_e at 0x7efe2e218400>,
 <Reaction EX_glcn_e at 0x7efe2e218520>,
 <Reaction BIOMASS_Ec_iJO1366_WT_53p95M at 0x7efe2e218640>,
 <Reaction EX_glcr_e at 0x7efe2e218760>,
 <Reaction EX_colipap_e at 0x7efe2e218a30>,
 <Reaction EX_glcur_e at 0x7efe2e2182e0>,
 <Reaction EX_glcur1p_e at 0x7efe2e2181c0>,
 <Reaction BIOMASS_Ec_iJO1366_core_53p95M at 0x7efe2e218490>,
 <Reaction EX_12ppd__R_e at 0x7efe2e2187f0>,
 <Reaction EX_gln__L_e at 0x7efe2e238b20>,
 <Reaction EX_cpgn_e at 0x7efe2e2387c0>,
 <Reaction EX_glu__L_e at 0x7efe2e2

In [127]:
print(model.objective)
model.optimize
model.optimize()

Maximize
1.0*BIOMASS_Ec_iJO1366_core_53p95M - 1.0*BIOMASS_Ec_iJO1366_core_53p95M_reverse_5c8b1


<Solution 0.982 at 0x7efe29e3f550>

In [128]:
with model:
    solution = model.optimize()
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Maximum theoretical biomass productivity: 0.9823718127269727 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 6.390286975717436 mmol/gDW*h


2 Level, 21 Factor Fractorial DOE is performed to determine best combination of amino acids in Media. Glucose is used as C-source at level 20 or 40. All 20 common AA are used at level 0 or 20.

Result:
Out of 20 AA, these 12 have the biggest influence:
Arg
Pro
Gln
Thr
Glu
Asp
Cys
Ala
Ser
Trp
Asn
Val
Gly

Glucose at 40 is better then at 20

Maximum theoretical biomass productivity was increased from 0.982 /h to 17.792 /h in the 'Statistically best' run. This is roughly a 18-times improvement.
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction was increased from 6.39 mmol/gDW*h to 116.167 mmol/gDW*h. This is roughly a 18 times improvement.

Making Media for Run 1 of 32

In [129]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 20

Productivity using Run 1 Media

In [130]:
with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Maximum theoretical biomass productivity: 9.278337088982106 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 49.563207547169874 mmol/gDW*h


Making Media for Run 2 of 32

In [131]:
YEPDmedium2 = model.medium
YEPDmedium2['EX_glc__D_e'] = 20
YEPDmedium2['EX_ala__L_e'] = 0
YEPDmedium2['EX_arg__L_e'] = 0
YEPDmedium2['EX_asn__L_e'] = 0
YEPDmedium2['EX_cys__L_e'] = 20
YEPDmedium2['EX_gln__L_e'] = 20
YEPDmedium2['EX_glu__L_e'] = 0
YEPDmedium2['EX_gly_e'] = 0
YEPDmedium2['EX_his__L_e'] = 20
YEPDmedium2['EX_ile__L_e'] = 0
YEPDmedium2['EX_leu__L_e'] = 20
YEPDmedium2['EX_met__L_e'] = 20
YEPDmedium2['EX_phe__L_e'] = 0
YEPDmedium2['EX_pro__L_e'] = 0
YEPDmedium2['EX_ser__L_e'] = 20
YEPDmedium2['EX_thr__L_e'] = 20
YEPDmedium2['EX_trp__L_e'] = 0
YEPDmedium2['EX_tyr__L_e'] = 20
YEPDmedium2['EX_val__L_e'] = 0
YEPDmedium2['EX_lys__L_e'] = 0
YEPDmedium2['EX_asp__L_e'] = 20

Productivity using Run 2 Media

In [132]:
with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Maximum theoretical biomass productivity: 9.278337088984305 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 49.56320754716974 mmol/gDW*h


Making Media for Run 3

In [133]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 0

Productivity for Run 3 Media

In [134]:
with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Maximum theoretical biomass productivity: 8.394074883348978 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 46.072641509434106 mmol/gDW*h


Media and Productivity for Run 4

In [135]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 4:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 4:
Maximum theoretical biomass productivity: 8.578553785827237 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 52.58207547169789 mmol/gDW*h


Media and Productivity Run 5:

In [136]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 5:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 5:
Maximum theoretical biomass productivity: 9.12315621113567 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 49.0915094339623 mmol/gDW*h


Media and Productivity for Run 6:

In [137]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 6:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 6:
Maximum theoretical biomass productivity: 10.044916712188819 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 59.37452830188613 mmol/gDW*h


Media and Productivity for Run 7:

In [138]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 7:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 7:
Maximum theoretical biomass productivity: 11.474637602157461 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 65.97830188679202 mmol/gDW*h


Media and Productivity for Run 8:

In [139]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 8:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 8:
Maximum theoretical biomass productivity: 11.839920431425965 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 69.28018867924516 mmol/gDW*h


Media and Productivity for Run 9:

In [140]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 9:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 9:
Maximum theoretical biomass productivity: 7.715016342916594 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 43.43113207547172 mmol/gDW*h


Media and Productivity for Run 10:

In [141]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 10:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 10:
Maximum theoretical biomass productivity: 9.588258989523469 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 53.52547169811315 mmol/gDW*h


Media and Productivity for Run 11:

In [142]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 11:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 11:
Maximum theoretical biomass productivity: 9.230116382842379 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 53.33679245282999 mmol/gDW*h


Media and Productivity for Run 12:

In [143]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 12:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 12:
Maximum theoretical biomass productivity: 12.26377917462079 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 70.41226415094323 mmol/gDW*h


Media and Productivity for Run 13:

In [144]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 13:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 13:
Maximum theoretical biomass productivity: 12.031926421216264 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 66.92169811320744 mmol/gDW*h


Media and Productivity for Run 14:

In [145]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 14:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 14:
Maximum theoretical biomass productivity: 11.718214527486746 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 66.63867924528306 mmol/gDW*h


Media and Productivity for Run 15:

In [146]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 15:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 15:
Maximum theoretical biomass productivity: 12.104389629283762 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 70.41226415094341 mmol/gDW*h


Media and Productivity for Run 16:

In [147]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 16:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 16:
Maximum theoretical biomass productivity: 11.202610270549926 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 63.1481132075471 mmol/gDW*h


Media and Productivity for Run 17:

In [148]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 17:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 17:
Maximum theoretical biomass productivity: 10.893551562846618 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 61.2198412698412 mmol/gDW*h


Media and Productivity for Run 18:

In [149]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 18:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 18:
Maximum theoretical biomass productivity: 9.165181952757706 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 50.197500000000055 mmol/gDW*h


Media and Productivity for Run 19:

In [150]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 19:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 19:
Maximum theoretical biomass productivity: 10.346251307554223 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 57.01349206349219 mmol/gDW*h


Media and Productivity for Run 20:

In [151]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 20:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 20:
Maximum theoretical biomass productivity: 13.478264575343326 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 81.26132075471688 mmol/gDW*h


Media and Productivity for Run 21:

In [152]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 21:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 21:
Maximum theoretical biomass productivity: 11.782936409853493 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 63.61981132075439 mmol/gDW*h


Media and Productivity for Run 22:

In [153]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 22:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 22:
Maximum theoretical biomass productivity: 14.67463709251048 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 84.46886792452833 mmol/gDW*h


Media and Productivity for Run 23:

In [154]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 23:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 23:
Maximum theoretical biomass productivity: 14.419707338906953 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 81.26132075471651 mmol/gDW*h


Media and Productivity for Run 24:

In [155]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 24:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 24:
Maximum theoretical biomass productivity: 11.46782593799311 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 66.8273584905661 mmol/gDW*h


Media and Productivity for Run 25:

In [156]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 25:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 25:
Maximum theoretical biomass productivity: 12.496338974960468 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 68.33679245283041 mmol/gDW*h


Media and Productivity for Run 26:

In [157]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 26:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 26:
Maximum theoretical biomass productivity: 12.050744002953202 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 68.24245283018868 mmol/gDW*h


Media and Productivity for Run 27:

In [158]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 27:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 27:
Maximum theoretical biomass productivity: 14.284918745090128 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 92.95943396226387 mmol/gDW*h


Media and Productivity for Run 28:

In [159]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 28:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 28:
Maximum theoretical biomass productivity: 8.885559238096903 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 43.53083333333314 mmol/gDW*h


Media and Productivity for Run 29:

In [160]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 29:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 29:
Maximum theoretical biomass productivity: 11.94537197811599 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 68.24245283018871 mmol/gDW*h


Media and Productivity for Run 30:

In [161]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 30:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 30:
Maximum theoretical biomass productivity: 14.591504738742781 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 78.14811320754696 mmol/gDW*h


Media and Productivity for Run 31:

In [162]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 31:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 31:
Maximum theoretical biomass productivity: 8.609986872263036 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 57.013492063491945 mmol/gDW*h


Media and Productivity for Run 32:

In [163]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 32:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 32:
Maximum theoretical biomass productivity: 19.36915342869905 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 116.16698113207559 mmol/gDW*h


Testing 'Statistically Best' Media:

In [164]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Statistically best:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Statistically best:
Maximum theoretical biomass productivity: 17.79244770765661 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 116.1669811320755 mmol/gDW*h


Statistically best best:

In [166]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Statistically best best:')
    print('Maximum theoretical biomass productivity:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Statistically best best:
Maximum theoretical biomass productivity: 18.251778743398255 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 116.16698113207575 mmol/gDW*h
